From 3ed7bc65808fbbab312e798607b92b7fe644ed34 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kacper=20Michaj=C5=82ow?= <kasper93@gmail.com>
Date: Wed, 14 Oct 2026 10:39:20 +0200
Subject: [PATCH] avformat/hls: keep the rendition dispositions when a stream
 is updated
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add the dispositions of the subdemuxer stream to the ones set from the
EXT-X-MEDIA attributes instead of replacing them.

Signed-off-by: Kacper Michajłow <kasper93@gmail.com>
(cherry picked from commit 3ed7bc65808fbbab312e798607b92b7fe644ed34)
---
 libavformat/hls.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libavformat/hls.c b/libavformat/hls.c
index 129df0ccff8..766f9a6274e 100644
--- a/libavformat/hls.c
+++ b/libavformat/hls.c
@@ -2359,8 +2359,8 @@ static int set_stream_info_from_input_stream(AVStream *st, struct playlist *pls,
     else
         avpriv_set_pts_info(st, ist->pts_wrap_bits, ist->time_base.num, ist->time_base.den);
 
-    // copy disposition
-    st->disposition = ist->disposition;
+    // add the dispositions of the subdemuxer stream to the rendition ones
+    st->disposition |= ist->disposition;
 
     av_dict_copy(&st->metadata, ist->metadata, 0);
 
-- 
2.53.0

